2-mandatory/6-journey-planner.js

```diff
@@ -122,7 +128,12 @@ function getLocationName() {}
   Advanced challange: try to use arrow function when invoking an array method.
 */
 function journeyPlanner(locations, transportMode) {
-  // Implement the function body
+  let finalArray =  locations.map(item=>{
+    let transModes = getTransportModes(item);
+    if (isAccessibleByTransportMode(transModes,transportMode)) 
+    return getLocationName(item).toString()
+  })
+  return finalArray.filter(element => {return element !== undefined});
 }
```

Comment:
I have declared more new variables inside the function to achieve the same result